Source/WebCore/platform/glib/PasteboardGLib.cpp

```diff
@@ -334,14 +356,33 @@ void Pasteboard::read(PasteboardFileReader& reader, std::optional<size_t> index)
 
     if (!index) {
         auto filePaths = platformStrategies()->pasteboardStrategy()->readFilePathsFromClipboard(m_name);
-        for (const auto& filePath : filePaths)
-            reader.readFilename(filePath);
+        if (filePaths.size() > 0) {
+            for (const auto& filePath : filePaths)
+                reader.readFilename(filePath);
+            return;
+        }
+    }
+
+    auto readBufferAtIndex = [&](size_t itemIndex) {
+        for (const auto& imageType : imageTypes()) {
+            if (!reader.shouldReadBuffer(imageType))
+                continue;
+            auto buffer = readBuffer(itemIndex, imageType);
+            if (buffer) {
+                reader.readBuffer(imageTypeToFakeFilename(imageType), imageType, buffer.releaseNonNull());
```

Comment:
Than can probably be:

```suggestion
                reader.readBuffer(imageTypeToFakeFilename(imageType), imageType, WTFMove(buffer));
```

I'll investigate before landing.

Reply:
No, it was right. I was afraid that `releaseNonNull()` was leaking, but it actually just converts RefPtr to Ref.